kinsumer.go

```diff
@@ -454,6 +463,8 @@ func (k *Kinsumer) Run() error {
 				if !k.config.manualCheckpointing {
 					record.checkpointer.update(aws.ToString(record.record.SequenceNumber))
 				}
+				atomic.AddInt64(&k.recordsInMemoryCount, -1)
+				atomic.AddInt64(&k.bytesInMemoryCount, -record.payloadBytes)
```

Comment:
Do we need an atomic process or can we just throw these values into channels that are then consumed out of the critical path? Worried this will add quite a bit of locking to what is currently a multi-threaded processing loop right?

Reply:
OK, pushed a refactor that I think resolves this, but keeps channel usage overhead to a minimum as well. LMK what you think!